r2/service.go

```diff
@@ -122,6 +125,62 @@ type r2Handler struct {
 	auth  auth.HTTPAuthService
 }
 
+type routeFunc func(s *service, r *http.Request) (data interface{}, err error)
+
+type serviceMetrics struct {
+	sync.RWMutex
+	scope tally.Scope
+
+	// Endpoints mapped to metrics (endpoint = path + method + status).
+	metrics map[string]serviceMetric
+}
+
+type serviceMetric struct {
```

Comment:
Perhaps define a method `Record` here:
```
func (m *serviceMetric) Record(d time.Duration) {
  m.timer.Record(d)
  m.counter.Inc(1)
}
```
so you can simply call `m.Record(d)` in `handleRoute`?